### ISSUE TITLE
New Data Frame Error

### ISSUE THREAD

**Issue by jkmusyoka** [IDEMSInternational/R-Instat#4276]
I could not create a new data frame of more than 1000 rows in R-Instat 0.4.9. The attached error popped up 

![newdataframeerror](https://user-images.githubusercontent.com/8112611/33421068-e939e422-d5c1-11e7-81ce-2dae7306260b.png)

**Comment by rdstern** [comment]
This was to generate a new random sample.  It used to work, but does not now.

**Comment by dannyparsons** [comment]
Is this the same issue as https://github.com/africanmathsinitiative/R-Instat-Help/issues/13?

**Comment by Patowhiz** [comment]
@dannyparsons I'm able to create a random sample sample of n=4000 from Experimental Survey Dataset. I'm trying to figure out why the user received the error in  [https://github.com/africanmathsinitiative/R-Instat-Help/issues/13](url) . Where can I get the  Rice_Yield_Population data set ? so that I can also use it to test this.

**Comment by rdstern** [comment]
This is an issue that has been raised elsewhere.  The bug is when you do File > New and create a new worksheet with **just a single column and more than 1000 rows**.
It works fine with 2 columns, etc and is just on this issue.  It is nothing to do with creating the random sample.

**Comment by Patowhiz** [comment]
@rdstern  there is an issue that has been raised here  [ africanmathsinitiative/R-Instat-Help #13](https://github.com/africanmathsinitiative/R-Instat-Help/issues/13) . I was avoiding to comment it there because its an R-Instant issue. So I commented in this issue because Danny referenced it.

**Comment by Patowhiz** [comment]
Nevertheless, I'm looking at both of them.

**Comment by Patowhiz** [comment]
Looking deeper into the problem of generating new data frame, it appears that the real problem is in generating a data frame of 1 column with >1000 rows.  If the number of columns are > 1 then the dialog data frame is generated without any error.

**Comment by dannyparsons** [comment]
Ok thanks for clarifying. That makes more sense, we've had issues with single column data frames before - usually related to it getting converted to a vector at some point. Following the error and seeing the error message from R should show what the cause is.

**Comment by Patowhiz** [comment]
This is interesting. I have alienated the problem here.
The data.frame  Rcommand works fine and it creates the data frame successfully. The issue arises when the grid is being updated.  The problem is in the `$get_data_frame()` Rcommand that which throws  Error in `if (nrow(data) == 0) { : argument is of length zero `
@africanmathsinitiative/developers we can discuss on how we can address this. The funny thing is this only happens when the no of column = 1 and convert_to_character=TRUE .

**Comment by dannyparsons** [comment]
This is likely to be because `data` is not a `data.frame` hence `nrow(data)` returns `NA` would be my guess.
I would guess that `get_data_frame` is converting the 1 column data frame to a vector - which can easily happen by accident in R. The solution is likely to be in the R code. I can take a look if needed.

**Comment by rdstern** [comment]
It is not a coincidence that we only display 1000 rows in the data frame and that is when it goes wrong.  If you increase the limit of the display to 4000 (say) then you can display a one-column data frame up to this new limit.

**Comment by Patowhiz** [comment]
True. And also its a good thing we have sorted this out for end users who
don't know the limitations of R

On 13 Dec 2017 16:12, "rdstern" <notifications@github.com> wrote:

It is not a coincidence that we only display 1000 rows in the data frame
and that is when it goes wrong. If you increase the limit of the display to
4000 (say) then you can display a one-column data frame up to this new
limit.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/africanmathsinitiative/R-Instat/issues/4276#issuecomment-351387100>,
or mute the thread

**Comment by dannyparsons** [comment]
This should now be fixed, can someone test it out.

**Comment by lilyclements** [comment]
Can confirm that this is fixed
